controller/src/main/java/io/pravega/controller/server/rpc/auth/RESTAuthHelper.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2019 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.rpc.auth;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import io.pravega.auth.AuthException;
+import io.pravega.auth.AuthHandler;
+import io.pravega.auth.AuthenticationException;
+import io.pravega.auth.AuthorizationException;
+
+import javax.ws.rs.core.Response;
+import java.security.Principal;
+import java.util.List;
+
+/**
+ * Helper class for handling auth (authentication and authorization) for the REST API.
+ */
+public class RESTAuthHelper {
+
+    /**
+     * The delegate used for performing authentication and authorization.
+     */
+    private final PravegaAuthManager pravegaAuthManager;
+
+    public RESTAuthHelper(PravegaAuthManager pravegaAuthManager) {
+        this.pravegaAuthManager = pravegaAuthManager;
+    }
+
+    /**
+     * Determines whether the given {@code principal} has specified {@code permission} on the given {@code resource}.
```

Comment:
To be consistent with the definition here, I want to confirm that we are using `subject` to denote the abstract entity being authorized and `principal` being the code object, is that it?